controllers/platform/services.go

```diff
@@ -0,0 +1,368 @@
+	"github.com/apache/incubator-kie-kogito-serverless-operator/container-builder/client"
+	"github.com/apache/incubator-kie-kogito-serverless-operator/controllers/profiles/common"
+	"github.com/apache/incubator-kie-kogito-serverless-operator/log"
+	"github.com/apache/incubator-kie-kogito-serverless-operator/utils"
+	kubeutil "github.com/apache/incubator-kie-kogito-serverless-operator/utils/kubernetes"
+	"github.com/apache/incubator-kie-kogito-serverless-operator/workflowproj"
+	"github.com/imdario/mergo"
+)
+
+// NewServiceAction returns an action that deploys the services.
+func NewServiceAction() Action {
+	return &serviceAction{}
+}
+
+type serviceAction struct {
+	baseAction
+}
+
+func (action *serviceAction) Name() string {
+	return "service"
+}
+
+func (action *serviceAction) CanHandle(platform *operatorapi.SonataFlowPlatform) bool {
+	return platform.Status.IsReady()
+}
+
+func (action *serviceAction) Handle(ctx context.Context, platform *operatorapi.SonataFlowPlatform) (*operatorapi.SonataFlowPlatform, error) {
+	// Refresh applied configuration
+	if err := ConfigureDefaults(ctx, action.client, platform, false); err != nil {
+		return nil, err
+	}
+
+	if err := createDataIndexComponents(ctx, action.client, platform); err != nil {
+		return nil, err
+	}
+
+	return platform, nil
+}
+
+func createDataIndexComponents(ctx context.Context, client client.Client, platform *operatorapi.SonataFlowPlatform) error {
+	if platform.Spec.Services.DataIndex != nil {
+		if err := createDataIndexConfigMap(ctx, client, platform); err != nil {
+			return err
+		}
+		if err := createDataIndexDeployment(ctx, client, platform); err != nil {
+			return err
+		}
+		if err := createDataIndexService(ctx, client, platform); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func createDataIndexDeployment(ctx context.Context, client client.Client, platform *operatorapi.SonataFlowPlatform) error {
+	readyProbe := &corev1.Probe{
+		ProbeHandler: corev1.ProbeHandler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path:   common.QuarkusHealthPathReady,
+				Port:   common.DefaultHTTPWorkflowPortIntStr,
+				Scheme: corev1.URISchemeHTTP,
+			},
+		},
+		InitialDelaySeconds: int32(15),
+		TimeoutSeconds:      int32(10),
+		PeriodSeconds:       int32(30),
+		SuccessThreshold:    int32(1),
+		FailureThreshold:    int32(3),
+	}
+	liveProbe := readyProbe.DeepCopy()
+	liveProbe.ProbeHandler.HTTPGet.Path = common.QuarkusHealthPathLive
+	dataDeployContainer := &corev1.Container{
+		Name:  common.DataIndexName,
+		Image: common.DataIndexImageBase + common.PersistenceTypeEphemeral,
+		Env: []corev1.EnvVar{
+			{
+				Name:  "KOGITO_DATA_INDEX_QUARKUS_PROFILE",
+				Value: "http-events-support",
+			},
+			{
+				Name:  "QUARKUS_HTTP_PORT",
+				Value: "8080",
+			},
+			{
+				Name:  "QUARKUS_HTTP_CORS",
+				Value: "true",
+			},
+			{
+				Name:  "QUARKUS_HTTP_CORS_ORIGINS",
+				Value: "/.*/",
+			},
+			{
+				Name:  "QUARKUS_KAFKA_HEALTH_ENABLE",
```

Comment:
This configuration does not have effect, the value quarkus.smallrye-health.check."io.quarkus.kafka.client.health.KafkaHealthCheck".enabled=false  in the properties is good already.